include/secp256k1_musig.h

```diff
@@ -7,348 +7,152 @@
- *  Out: combined_pk: the MuSig-combined xonly public key (cannot be NULL)
- *       pre_session: if non-NULL, pointer to a musig_pre_session struct to be used in
- *                    `musig_session_init` or `musig_pubkey_tweak_add`.
- *   In:     pubkeys: input array of pointers to public keys to combine. The order
- *                    is important; a different order will result in a different
- *                    combined public key (cannot be NULL)
- *         n_pubkeys: length of pubkeys array. Must be greater than 0.
+ *  Guaranteed to be 132 bytes in size. Serialized and parsed with
+ *  `musig_aggnonce_serialize` and `musig_aggnonce_parse`.
  */
-SECP256K1_API int secp256k1_musig_pubkey_combine(
-    const secp256k1_context* ctx,
-    secp256k1_scratch_space *scratch,
-    secp256k1_xonly_pubkey *combined_pk,
-    secp256k1_musig_pre_session *pre_session,
-    const secp256k1_xonly_pubkey * const* pubkeys,
-    size_t n_pubkeys
-) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(5);
+typedef struct {
+    unsigned char data[132];
+} secp256k1_musig_aggnonce;
 
-/** Tweak an x-only public key by adding the generator multiplied with tweak32
- *  to it. The resulting output_pubkey with the given internal_pubkey and tweak
- *  passes `secp256k1_xonly_pubkey_tweak_test`.
- *
- *  This function is only useful before initializing a signing session. If you
- *  are only computing a public key, but not intending to create a signature for
- *  it, you can just use `secp256k1_xonly_pubkey_tweak_add`. Can only be called
- *  once with a given pre_session.
+/** Opaque data structure that holds a cache for a MuSig session.
  *
- *  Returns: 0 if the arguments are invalid or the resulting public key would be
- *           invalid (only when the tweak is the negation of the corresponding
- *           secret key). 1 otherwise.
- *  Args:          ctx: pointer to a context object initialized for verification
- *                      (cannot be NULL)
- *         pre_session: pointer to a `musig_pre_session` struct initialized in
- *                      `musig_pubkey_combine` (cannot be NULL)
- *  Out: output_pubkey: pointer to a public key to store the result. Will be set
- *                      to an invalid value if this function returns 0 (cannot
- *                      be NULL)
- *  In: internal_pubkey: pointer to the `combined_pk` from
- *                       `musig_pubkey_combine` to which the tweak is applied.
- *                       (cannot be NULL).
- *              tweak32: pointer to a 32-byte tweak. If the tweak is invalid
- *                       according to secp256k1_ec_seckey_verify, this function
- *                       returns 0. For uniformly random 32-byte arrays the
- *                       chance of being invalid is negligible (around 1 in
- *                       2^128) (cannot be NULL).
+ *  This structure is not necessarily required to be kept secret. Guaranteed to
```

Comment:
Is it possible to be explicit about when it needs to be kept secret, or otherwise just say "it's not required to be kept secret"?

Reply:
Rephrased